File: solutions/1701. Average Waiting Time/1701.py
# Time complexity: O(n)
# Space complexity: O(1)
class Solution:
  def averageWaitingTime(self, customers: List[List[int]]) -> float:
    wait = 0
    curr = 0

    for customer in customers:
      curr = max(curr, 1.0 * customer[0]) + customer[1]
      wait += curr - customer[0]

    return 1.0 * wait / len(customers)

File: solutions/1701. Average Waiting Time/1701.cpp
// Time complexity: O(n)
// Space complexity: O(1)
class Solution {
 public:
  double averageWaitingTime(const vector<vector<int>>& customers) {
    double wait = 0;
    double curr = 0;

    for (const vector<int>& customer : customers) {
      curr = max(curr, 1.0 * customer[0]) + customer[1];
      wait += curr - customer[0];
    }

    return 1.0 * wait / customers.size();
  }
};